src/python/grpcio/grpc/_channel.py

```diff
@@ -374,13 +376,31 @@ def details(self):
                 self._state.condition.wait()
             return _common.decode(self._state.details)
 
+    def debug_error_string(self):
+        with self._state.condition:
+            while self._state.debug_error_string is None:
+                self._state.condition.wait()
+            return _common.decode(self._state.debug_error_string)
+
+    def _format_debug_error_string(self):
+      return self._state.debug_error_string
+
+    def _format_completed_repr(self):
+        string = "<_Rendezvous of RPC that terminated with:\n"
```

Comment:
Use a
```python
_OK_RENDEZVOUS_REPR_FORMAT = (
    '<_Rendezvous of RPC that terminated with:\n'
    '\tstatus = {}\n'
    '\tdetails = "{}"\n'
    '>')
```
constant and a separate `_NON_OK_RENDEZVOUS_REPR_FORMAT` constant?

[Definitely don't ever build strings with `+=`.](https://www.joelonsoftware.com/2001/12/11/back-to-basics/) Sometimes `+`, but never `+=`.

Reply:
Done